beacon-chain/rpc/eth/validator/handlers_block.go

```diff
@@ -21,7 +21,119 @@ import (
 	"go.opencensus.io/trace"
 )
 
-// ProduceBlockV3 Requests a beacon node to produce a valid block, which can then be signed by a validator. The
+type blockType uint8
+
+const (
+	any blockType = iota
+	full
+	blinded
+)
+
+// DEPRECATED: Please use ProduceBlockV3 instead.
+//
+// ProduceBlockV2 requests the beacon node to produce a valid unsigned beacon block,
+// which can then be signed by a proposer and submitted.
+func (s *Server) ProduceBlockV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ProduceBlockV2")
+	defer span.End()
+
+	if shared.IsSyncing(ctx, w, s.SyncChecker, s.HeadFetcher, s.TimeFetcher, s.OptimisticModeFetcher) {
+		return
+	}
+
+	segments := strings.Split(r.URL.Path, "/")
+	rawSlot := segments[len(segments)-1]
+	rawRandaoReveal := r.URL.Query().Get("randao_reveal")
+	rawGraffiti := r.URL.Query().Get("graffiti")
+	rawSkipRandaoVerification := r.URL.Query().Get("skip_randao_verification")
+
+	slot, valid := shared.ValidateUint(w, "slot", rawSlot)
+	if !valid {
+		return
+	}
+
+	var randaoReveal []byte
+	if rawSkipRandaoVerification == "true" {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := shared.DecodeHexWithLength(rawRandaoReveal, fieldparams.BLSSignatureLength)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode randao reveal").Error(), http.StatusBadRequest)
```

Comment:
If I recall correctly we decided to capitalize all http errors
```suggestion
			http2.HandleError(w, errors.Wrap(err, "Unable to decode randao reveal").Error(), http.StatusBadRequest)
```